assets/js/app.js

```diff
@@ -0,0 +1,390 @@
+      });
+    });
+  };
+  try {
+    // allCoins = await getData("assets/json/file.json"); //working locale
+    allCoins = await getData("https://api.coingecko.com/api/v3/coins/list"); // working with api
+
+    const firstHundredCoins = allCoins.slice(0, 100);
+    const newHTML = await getCoinHTML(firstHundredCoins);
+    renderAllCoins(newHTML);
+    hideLoader();
+
+    document.getElementById("searchBtn").addEventListener("click", (event) => {
+      event.preventDefault();
+      let searchInput = document.getElementById(`searchInput`);
+      handleSearch(searchInput.value, allCoins);
+      searchInput.value = "";
+    });
+  } catch (e) {
+    console.warn(e);
+    const myToast = new bootstrap.Toast(document.getElementById("myToast"));
+    myToast.show();
+    setTimeout(() => {
+      location.reload();
+    }, 60000);
+  }
+
+  document.getElementById("profile-tab").addEventListener("click", async () => {
+    window.xValues = [];
+    window.datasetsMap = {};
+
+    const initialObjectCoinGraphData = await fetchCoinGraphData();
+    CoinGraphData(initialObjectCoinGraphData);
+    setInterval(async () => {
+      const objectCoinGraphData = await fetchCoinGraphData();
+      CoinGraphData(objectCoinGraphData);
+    }, 5000);
+  });
+  const getCurrenciesSymbols = () => {
+    const selectedCoins =
+      JSON.parse(localStorage.getItem("selectedCoins")) || [];
+    return selectedCoins.map((coin) => coin.symbol).join(",");
+  };
+
+  const CoinGraphData = (objectCoinGraphData) => {
+    if (!window.xValues) window.xValues = [];
+    if (!window.datasetsMap) window.datasetsMap = {};
+    const currentTime = new Date().toLocaleTimeString();
+    window.xValues.push(currentTime);
+    Object.entries(objectCoinGraphData).forEach(([symbol, data]) => {
+      const { USD } = data;
+
+      if (!window.datasetsMap[symbol]) {
+        window.datasetsMap[symbol] = [];
+      }
+
+      window.datasetsMap[symbol].push(USD);
+    });
+
+    const getColor = (index) => {
+      const colors = ["red", "green", "blue", "orange", "purple"];
+      return colors[index % colors.length];
+    };
+
+    const newDatasets = Object.keys(window.datasetsMap).map(
+      (symbol, index) => ({
+        label: symbol.toUpperCase(),
+        data: window.datasetsMap[symbol],
+        borderColor: getColor(index),
+        fill: false,
+      })
+    );
+    if (newDatasets.length >= 7) {
+      const myToast = new bootstrap.Toast(
+        document.getElementById("graphToast")
+      );
+      myToast.show();
+    } else {
+      if (window.chart) {
+        window.chart.data.labels = window.xValues;
+        window.chart.data.datasets = newDatasets;
+        window.chart.update();
+      } else {
+        window.chart = new Chart("myChart", {
+          type: "line",
+          data: {
+            labels: window.xValues,
+            datasets: newDatasets,
+          },
+        });
+      }
+    }
+  };
+
+  const fetchCoinGraphData = async () =>
```

Comment:
this function should have been broken to at least 2. you should always separate logic code from presentation code.